go/cli/mcap/cmd/info.go

```diff
@@ -238,8 +238,22 @@ func printInfo(w io.Writer, info *mcap.Info) error {
 		if info.Statistics != nil {
 			channelMessageCount := info.Statistics.ChannelMessageCounts[chanID]
 			if channelMessageCount > 1 {
-				frequency := 1e9 * float64(channelMessageCount) / float64(end-start)
-				row = append(row, fmt.Sprintf("%*d msgs (%.2f Hz)", maxCountWidth, channelMessageCount, frequency))
+				// Estimate frequency from statistics.
+				// This assumes the underlying channel is logged at a quasi-constant rate with
+				// random jitter.  Since we don't know where the MCAP start and end land in the
+				// message's period, we calculate an upper bound (assuming the MCAP ends just
+				// before the next message is logged) and a lower bound (assuming the MCAP ends
+				// immediately after the last message was logged).
+				// NOTE: We could make a better estimate by seeking to and inspecting message
+				// indexes. However, these aren't present in all MCAPs, and seeking to each one can
+				// be prohibitively slow, especially when accessing a remote file.
+				seconds := float64(end-start) / 1e9
+				maxHz := float64(channelMessageCount) / seconds
+				minHz := float64(channelMessageCount-1) / seconds
+				row = append(
+					row,
+					fmt.Sprintf("%*d msgs [%.2fHz..%.2fHz]", maxCountWidth, channelMessageCount, minHz, maxHz),
```

Comment:
I wonder if we should reduce the precision to `.1f` since we are now displaying a range like this? I guess ideally we could dynamically adjust the precision based on the magnitude of the difference, but that sounds harder and changing to `.1f` is easy.

Reply:
https://go.dev/play/p/zVmOESke55L